### PR TITLE
Import and refactor of grafeas-elasticsearch index manager

### PR DESCRIPTION
**Pull request by alexashley**
This is mostly a lift and shift of the code from `grafeas-elasticsearch`, with some restructuring to allow for consumers to reference a single interface when using the library. Previously, the `IndexManager` and `MigrationOrchestrator` were both used independently; they're still separate but the orchestrator is embedded in the `IndexManager`, which now exposes a function for loading mappings and running migrations. 

I also changed how the mappings are loaded, taking advantage of [Go 1.16's `fs.FS` interface](https://golang.org/doc/go1.16#fs) in order to make that code more testable.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+project_name: es-index-manager
+builds:
```
Comment (alexashley) [inline .goreleaser.yml:2]: This should configure goreleaser to create a GitHub release and changelog, but not build any artifacts. 

https://goreleaser.com/cookbooks/release-a-library/

File: indexmanager/registry.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	"path/filepath"
+	"strings"
+)
+
+//counterfeiter:generate -o ../mocks . MappingsRegistry
+type MappingsRegistry interface {
+	// LoadMappings reads the index mapping and version from JSON file in the directory specified in Config.MappingsPath.
+	LoadMappings() error
+	// IndexName returns the full index name, using the prefix, version, inner name, and document kind.
+	IndexName(documentKind, inner string) string
```
Comment (alexashley) [inline indexmanager/registry.go:30]: `inner`/`innerName` is the best I could come up with for the part of the index name that's not the prefix, version, or document kind.

File: indexmanager/index_manager.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	Initialize(context.Context) error
+}
+
+type indexManager struct {
+	MappingsRegistry
+	IndexRepository
+	MigrationOrchestrator
+}
+
+func NewIndexManager(logger *zap.Logger, client *elasticsearch.Client, config *Config) IndexManager {
```
Comment (mrparkers) [inline indexmanager/index_manager.go:46]: I think I like this approach better than the traditional "do all setup in `main.go`", which I suppose is even more important when this is meant to be consumed as a library.  I guess the only trade off is that this code won't be covered by unit tests, but I think that's worth it.

Comment (alexashley) [reply indexmanager/index_manager.go:46]: As another approach, I played around with [`fx`](https://github.com/uber-go/fx) a while ago and liked that quite a bit. It composes dependencies at runtime, so there's a hit to startup, but it worked pretty well.

Comment (mrparkers) [reply indexmanager/index_manager.go:46]: Nice, I haven't tried that yet.  the only other candidate we've tried in the past is [wire](https://github.com/google/wire), but we weren't a fan of that.  I'd be interested to see what `fx` looks like though.

**Review by mrparkers — APPROVED** [review]

**Pull request merged**